Services/ExerciseService.cs

```diff
@@ -23,6 +23,11 @@ public async Task<IEnumerable<Exercise>> GetExercises()
             return await _context.Exercises.Where(e => e.Id == exerciseId).FirstOrDefaultAsync();
         }
 
+        public async Task<Exercise?> GetExerciseByName(string name)
+        {
+            return await _context.Exercises.Where(e => e.Name == name).FirstOrDefaultAsync();
```

Comment:
you may want to think about case sensitivity here and try to match on a tolower of the string